iroh-net/src/magicsock.rs

```diff
@@ -400,6 +409,24 @@ impl MagicSock {
             .instrument(info_span!("derp.actor")),
         );
 
+        // load the peer data
+        let peer_map = match peers_path.as_ref() {
+            Some(path) if path.exists() => {
+                match PeerMap::load_from_file(path, actor_sender.clone()) {
```

Comment:
maybe just add a bulk add addr method?

Reply:
the endpoint is passed because the value is stored for each endpoint, but it's not used and peers are not sent via this (or any channel) so I don't think there is possibility of deadlock. 

This seems dependent on trusting that the inner code does not use the channel, but I think the sync/non async coloring saves us here:  `load_from_file` is non-async, so any async call would need to be spawned outside this(effectively doing what you suggest). Any other non-async call of the `try_send` fashion would fail in that scenario. So I think this is safe